packages/engine/src/scene/systems/ShadowSystem.ts

```diff
@@ -223,34 +237,38 @@ export default async function ShadowSystem(world: World) {
       for (const entity of dropShadowComponentQuery()) {
         const dropShadowComponent = getComponent(entity, DropShadowComponent)
 
-        if (!dropShadowComponent.center) continue
-
-        const groupComponent = getComponent(entity, GroupComponent)
+        if (!entityExists(dropShadowComponent.entity)) {
+          removeComponent(entity, DropShadowComponent)
+        }
 
         raycaster.firstHitOnly = true
-        raycasterPosition.copy(dropShadowComponent.center!)
-        groupComponent[0].localToWorld(raycasterPosition)
+        raycasterPosition.copy(dropShadowComponent.center)
+        getComponent(entity, GroupComponent)[0].localToWorld(raycasterPosition)
         raycaster.set(raycasterPosition, shadowDirection)
 
         const intersected = raycaster.intersectObjects(sceneObjects)[0]
         if (!intersected || !intersected.face) {
-          setDropShadowMatrix(defaultShadowMatrix)
+          setDropShadowMatrix(defaultShadowMatrix, dropShadowComponent.entity, entity)
           continue
         }
 
+        const halfPoint = 0.5
+        const centerCorrectedDist = Math.max(intersected.distance - dropShadowComponent.center.y * halfPoint, 0)
```

Comment:
True center would not be multiplied by a "halfpoint", this is a stopgap measure because casting from the true bottom of objects would give the avatar a tiny shadow. Going to fix possibly by adding the distance scaling to the start radius.